source/common/event/dispatcher_impl.cc

```diff
@@ -25,6 +25,10 @@
 
 #include "event2/event.h"
 
+#ifdef ENVOY_HANDLE_SIGNALS
+#include "exe/signal_action.h"
```

Comment:
Reaching into exe here seems wrong from a layering perspective. Would it be better to have some other type of injected interface for this purpose?

Reply:
for layering, honestly I'd think all of the signal handling code could be moved out of exe since we use it in test, so it's somewhat core to the server handling.   maybe source/server/signal/?

I'm also not sure if that addresses your concerns, or if you still think there's value in the dispatcher not reaching directly to signal handling code.

Reply:
sgtm regarding moving. I might even go as far as to move some of it into source/common/signal but up to you.